plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/lookup/MenuLookup.java

```diff
@@ -124,7 +122,47 @@ public ActionContributionItem run() {
 		});
 		return contItem;
 	}
+	
+	public MenuItem lookForViewMenu(final List<IContributionItem> contItems, final Matcher<String>... matchers) {
+		IWorkbenchPart part = getActivePart(false);
+		final IMenuManager m = ((IViewSite) part.getSite()).getActionBars().getMenuManager();
+		if (!(m instanceof MenuManager)) {
+			throw new CoreLayerException("No Menu found in active part: " + part.getTitle());
+		}
+		MenuItem contItem = Display.syncExec(new ResultRunnable<MenuItem>() {
 
+			@Override
+			public MenuItem run() {
+				MenuItem currentItem = null;
+				Menu currentMenu = ((MenuManager) m).getMenu();
+				for (Matcher<String> matcher : matchers) {
+					for (MenuItem item : currentMenu.getItems()) {
+						String normalized = item.getText().replace("&", "");
+						if (matcher.matches(normalized)) {
+
+							// is submenu?
+							if (item.getMenu() != null) {
+
+								// populate submenu
+								sendHide(item.getMenu(), true);
+								sendShowUI(item.getMenu());
+
+								// drill down
+								currentMenu = item.getMenu();
+								break;
+							} else {
+								
+								// we got a match
+								currentItem = item;
```

Comment:
how about adding a break here ? No need to continue in loop when we already have a match